internal/policy/container/has_prohibited_labels.go

```diff
@@ -0,0 +1,65 @@
+package container
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/check"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/image"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/log"
+
+	"github.com/go-logr/logr"
+)
+
+var trademarkLabels = []string{"name", "vendor", "maintainer"}
+
+var _ check.Check = &HasProhibitedLabelsCheck{}
+
+type HasProhibitedLabelsCheck struct{}
+
+func (p *HasProhibitedLabelsCheck) Validate(ctx context.Context, imgRef image.ImageReference) (result bool, err error) {
+	labels, err := getDataForValidate(imgRef.ImageInfo)
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve image labels: %v", err)
+	}
+
+	return p.validate(ctx, labels)
+}
+
+func (p *HasProhibitedLabelsCheck) validate(ctx context.Context, labels map[string]string) (bool, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+
+	trademarkViolationLabels := []string{}
+	for _, label := range trademarkLabels {
+		if violatesRedHatTrademark(labels[label]) {
+			trademarkViolationLabels = append(trademarkViolationLabels, label)
+		}
+	}
+
+	// TODO: We should be reporting this in the results, not in a log message
```

Comment:
Does this TODO need addressing before merge?

Reply:
No, this TODO is there since the beginning, just carrying it over to this check.